src/lib/divider/divider.ts

```diff
@@ -0,0 +1,55 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {
+  ChangeDetectionStrategy, Component,
+  Directive,
+  Input, ViewEncapsulation,
+} from '@angular/core';
+import {coerceBooleanProperty} from '@angular/cdk/coercion';
+
+@Component({
+  moduleId: module.id,
+  selector: 'mat-divider',
+  host: {
+    'role': 'separator',
+    '[attr.aria-orientation]': 'vertical ? "vertical" : "horizontal"',
+    '[class.mat-divider-vertical]': 'vertical',
+    '[class.mat-divider-inset]': 'inset',
+  },
+  template: '<ng-content></ng-content>',
+  styleUrls: ['divider.css'],
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  preserveWhitespaces: false,
+})
+export class MatDivider {
+  /** Whether the divider is vertically aligned. */
+  @Input() get vertical(): boolean { return this._vertical; }
+  set vertical(value: boolean) {
+    this._vertical = coerceBooleanProperty(value);
+  }
+  protected _vertical: boolean = false;
+
+  /** Whether the divider is an inset divider. */
+  @Input() get inset(): boolean { return this._inset; }
+  set inset(value: boolean) {
+    this._inset = coerceBooleanProperty(value);
+  }
+  protected _inset: boolean = false;
+}
+
+/**
+ * Directive whose purpose is to add the mat- CSS styling to this selector.
+ * @docs-private
+ */
+@Directive({
+  selector: 'mat-divider',
+  host: {'class': 'mat-divider'}
+})
+export class MatDividerCssMatStyler {}
```

Comment:
I am not sure I see why this directive needs to exist.

Since it uses the same selector as `MatDivider` it will trigger on the same elements.  You can just add `'class': 'mat-divider'` to the `host` attribute in the `@Component` decorator for `MatDivider`

Reply:
This is a layover from the list component, pulled verbatim. Integrated with other component as suggested.